app/models/driver.rb

```diff
@@ -0,0 +1,33 @@
+class Driver < ApplicationRecord
+  has_many :trips, dependent: :nullify
+
+  validates_presence_of :name, :vin
+
+  def average_rating
+    counter = 0
+    rating = 0
+
+    trips.each do |trip|
+      if trip.rating != nil && trip.rating !=0
+        rating += trip.rating
+        counter += 1
+      end
+    end
+
+    return (rating.to_f / counter)
```

Comment:
What happens if there are no trips associated with the driver? Then this above line will execute with counter = 0 and rating = 0. In Ruby, `0.0/0` evaluates to `NaN` (Not a Number), so in the driver show page, if there are no trips associated with the driver, then `NaN.to_s` will evaluate to the string "NaN" and is shown on the page.